circuits/test/gfmulint/gfmulint.test.ts

```diff
@@ -21,31 +19,16 @@ describe("gfmulint", () => {
   });
 
   it("should output correct gfmul", async () => {
-    const a = 128;
-    const b = 128;
-    const expected = a * b;
-    const input = { a: pad_num_to_arr128(a), b: pad_num_to_arr128(b) };
+    /// x^128 = x^7 + x^2 + x + 1 -> 10000111
+    // Construct a 128-bit array with only the leftmost (most significant) bit set
+    const a_bits = new Array(128).fill(0);
+    a_bits[0] = 1;
 
-    let _res = await circuit.compute(input, ["res"]);
-    console.log(`res: ${_res.res}`);
-    let result = parse_arr128_to_number(_res.res as Arr128);
-    console.log(`${a} x ${b} = ${result}`);
-    assert.equal(result, expected);
-  });
-});
-
-function pad_num_to_arr128(value: number): Arr128 {
-  let tmp = value
-    .toString(2)
-    .padStart(128, "0")
-    .split("")
-    .map((bit) => parseInt(bit, 10));
-  return [tmp.slice(0, 64), tmp.slice(64, 128)];
-}
+    /// ( x^7 + x^2 + x + 1 ) * ( x^7 + x^2 + x + 1 ) = x^14 + x^4 + x^2 + 1 in this extension field
+    /// You can verify this in wolfwram alpha over the real numbers and then taking the resulting coefficients mod 2
+    /// x^14 + x^4 + x^2 + 1 -> 10001010000010 is the right most 15 bits of the expected result
+    let expected_bit_array = [1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0];
 
-function parse_arr128_to_number(res: Arr128): number {
-  let first_64: number[] = res[0];
-  let last_64: number[] = res[1];
-  let all_bits: number[] = first_64.concat(last_64);
-  return parseInt(all_bits.join(""), 2);
-}
+    await circuit.expectPass( { a: a_bits, b: a_bits }, { res: expected_bit_array });
```

Comment:
I think that using `compute` over `expectPass` is better for interpretability of the results and is in line with expected behavior of testing frameworks that will say something like:
```
test failed
expect: foo
got: bar
```

Reply:
i will probably close the pr for now since tackling the issue isn't ideal anymore

Reply:
oh why not? I likeed the CI